hibike/hibike_simulator.py

```diff
@@ -1,133 +1,144 @@
-                self.subscribedTo[UID]['flip_time'] = self.__calculateFlipTime(UID)
-            elif UID in self.UIDs and  UID in self.subscribedTo:
-                self.subscribedTo[UID]['delay'] = delay
-        return 0
-
-    def getData(self, UID, param):
-        """ Extracts all data for specific UID. Checks whether to update data,
-        flip data, or return previous data, and returns correct appropriate
-        data.
-        """
-
-        #TODO: param is currently unused
-
-        delay = self.subscribedTo[UID]['delay']
-        last_time = self.subscribedTo[UID]['time']
-        flip_time = self.subscribedTo[UID]['flip_time']
-        curr_time = time.time()*1000
-
-        should_flip = curr_time - delay >= flip_time
-        should_update = curr_time - delay >= last_time
-
-        if should_flip and should_update:
-            self.subscribedTo[UID]['time'] = curr_time
-            self.subscribedTo[UID]['data'] =  self.__getRandomData(UID, self.subscribedTo[UID]['is_low'])
-            self.subscribedTo[UID]['flip_time'] = self.__calculateFlipTime(UID)
-            self.subscribedTo[UID]['is_low'] = not self.subscribedTo[UID]['is_low']
-        elif should_update:
-            self.subscribedTo[UID]['time'] = curr_time
-            self.subscribedTo[UID]['data'] = self.__getRandomData(UID, not self.subscribedTo[UID]['is_low'])
-        return self.subscribedTo[UID]['data'] #returns sensor data
-
-    def __calculateFlipTime(self, UID):
-        """ Determines time of flip. """
-
-        rand, last_time = random.random(), self.subscribedTo[UID]['time']
-        noise = random.random()
-        for i in range(len(cutoffs)):
-            if rand < cutoffs[i]:
-                return (time.time() + noise + i)*1000
-
-    def __getRandomData(self, UID, is_low):
-        """ Finds device_type of UID and returns corresponding flipped data."""
-
-        device_type = UID[:6]
-        low, high, noise = sensor_values[device_type]
-        if low is None or high is None or noise is None:
-            return 0 # TODO: what to do with these device types?
-        if is_low:
-            return high + (random.random() - .5) * noise
-        else:
-            return low + (random.random() - .5) * noise
-
-############
-## MOTORS ##
-############
-
-    def readValue(self, UID, param):
-        #param values: 0 - delay, 1 - last_time, 2 - data
-        try:
-            return self.subscribedTo[UID][param]
-        except:
-            return 1
-
-    def writeValue(self, UID, param, value):
-        if param >= len(self.subscribedTo[UID]):
-            return 1
-        self.subscribedTo[UID][param] = value
-        return 0
+import threading
+import multiprocessing
+import hibike_message as hm
+import queue
+
+__all__ = ["hibike_process"]
+
+fake_uids = [0 << 72, 7 << 72]
+
+
+uid_to_index = {}
+
+
+def hibike_process(badThingsQueue, stateQueue, pipeFromChild):
+    while pipeFromChild.recv()[0] != "ready":
```

Comment:
What is the status regarding the "ready" message, which is supposedly deprecated?